### PR TITLE
Module SP Edit

### PR DESCRIPTION
**Pull request by allisandrana**
Update and Fix Stored Procedure And Trigger for Edit Laporan (and Deleting Laporan)

### CODE REVIEW

File: migrations/2020-11-04-070016_sp_laporan/up.sql [hunk 1]
```diff
@@ -54,19 +55,121 @@ $$ LANGUAGE plpgsql;
...
+RETURNS void AS $$
+DECLARE	
+	data_laporan record;
+	_urutan int;
+BEGIN
+	FOR data_laporan IN SELECT * FROM laporan WHERE satker_id = _id_satker AND jenis_id = _id_jenis AND EXTRACT(MONTH FROM tanggal_laporan) = EXTRACT(MONTH FROM _tanggal_laporan) AND EXTRACT(YEAR FROM tanggal_laporan) = EXTRACT(YEAR FROM _tanggal_laporan) AND tanggal_laporan >= _tanggal_laporan AND status = 'terkirim_sudah_diapprove' ORDER BY tanggal_laporan ASC LOOP
+		_urutan := num_count + 1;
+		UPDATE laporan SET nomor = sp_laporan_get_nomor(num_count,sp_laporan_get_kode_satker(_id_satker),sp_laporan_get_kode_referensi(_id_jenis),data_laporan.tanggal_laporan), urutan = _urutan WHERE id = data_laporan.id;
+		num_count := _urutan;
+	END LOOP;
```
Comment (izuleon) [inline migrations/2020-11-04-070016_sp_laporan/up.sql]:
```suggestion
BEGIN
	FOR data_laporan IN SELECT * 
	FROM laporan 
	WHERE 
	satker_id = _id_satker 
	AND jenis_id = _id_jenis 
	AND EXTRACT(MONTH FROM tanggal_laporan) = EXTRACT(MONTH FROM _tanggal_laporan) 
	AND EXTRACT(YEAR FROM tanggal_laporan) = EXTRACT(YEAR FROM _tanggal_laporan) 
	AND tanggal_laporan >= _tanggal_laporan 
	AND status = 'terkirim_sudah_diapprove' ORDER BY tanggal_laporan ASC LOOP
		_urutan := num_count + 1;
		UPDATE laporan SET nomor = sp_laporan_get_nomor(num_count,sp_laporan_get_kode_satker(_id_satker),sp_laporan_get_kode_referensi(_id_jenis),data_laporan.tanggal_laporan), urutan = _urutan WHERE id = data_laporan.id;
		num_count := _urutan;
	END LOOP;
```

Comment (izuleon) [reply migrations/2020-11-04-070016_sp_laporan/up.sql]: di rapihin dikit li boleh? biar urg bisa ikutan paham

Comment (allisandrana) [reply migrations/2020-11-04-070016_sp_laporan/up.sql]: ok

**Review by izuleon — APPROVED** [review]
LGTM

**Review by Ujang360 — APPROVED** [review]
Please carry on

**Pull request merged**